akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala

```diff
@@ -348,14 +356,17 @@ private abstract class RestartWithBackoffLogic[S <: Shape](
     sourceOut
   }
 
-  // Set a timer to restart after the calculated delay
-  protected final def scheduleRestartTimer() = {
+  protected final def maxRestartsReached = {
```

Comment:
As it also side effects, please add an empty parameter list to the method

Reply:
@johanandren is the ultimate authority here, I'm just Mr Style ;-) So let there be parenthesis in order to proceed.

Reply:
Rule of thumb is empty parameter list to hint that something isn't pure. Makes things a bit easier to reason about.

Reply:
What’s pure?

Reply:
As in not doing any side effects.

Reply:
✅ added the empty parameter list